36954 tags docs (#6622)
camunda/camunda-docs: docs/components/concepts/job-workers.md

- **Custom headers**: Additional static metadata that is defined in the process. Custom headers are used to configure reusable job workers (e.g. a `notify Slack` worker might read out the Slack channel from its header.)
- **Key**: Unique key to identify a job. The key is used to hand in the results of a job execution, or to report failures during job execution.
- **Variables**: The contextual/business data of the process instance required by the worker to do its work.

> - **Tags**: - **Tags**: Immutable labels copied from the process instance at job creation; Great for providing additional metadata (e.g., `reference:1234`, `team:accounting`, `trace-id:3004`). See [tags](#tags) and [process instance creation tags](/components/concepts/process-instance-creation.md#tags).

Reviewer: ```suggestion
- **Tags**: Immutable labels copied from the process instance at job creation; Great for providing additional metadata (e.g., `reference:1234`, `team:accounting`, `trace-id:3004`). See [tags](#tags) and [process instance creation tags](/components/concepts/process-instance-creation.md#tags).
```

For example, if jobs of a given type are not activated, but a worker is opened for this type, you can verify first if it exists in one of the gateways as a client stream. Once you've found it, grab its ID, and verify that you can find it as a consumer of a remote stream on each broker.

If it's not present in the gateway as a client stream, restart your worker. If it's not present as a consumer in one of the brokers, this indicates a bug. As a workaround, restart your gateway, which will cause some interruption in your service, but will force all streams for this gateway to be recreated properly.

## Tags (8.8+)

Tags are copied to each job exactly once: when the BPMN element is activated and the job is created. The job then holds an immutable snapshot of the process instance's tag set.

Key points:

- Tags are case-sensitive and immutable. They cannot be added, modified, or removed on the job.
- The tags on a job mirror the process instance's tags at job creation time and can not be altered.
- Apply tag-based logic inside the worker after activation, e.g. routing, prioritization, data correlation.
 

> See, use-cases [process instance creation tags](/components/concepts/process-instance-creation.md#tags-88) for more information (e.g. format, validation rules, limits, use-cases and guidelines).

Reviewer: ```suggestion
For detailed information about tag formats, validation rules, limits, and additional use cases, see [process instance creation tags](/components/concepts/process-instance-creation.md#tags-88).
```

| Tag length | 1–100 characters |
| Regex | `^[A-Za-z][A-Za-z0-9_\-:.]{0,99}$` |
| First character | Letter (A–Z / a–z) |
| Allowed subsequent characters | Alphanumeric, `_`, `-`, `:`, `.` |
| Case sensitivity | Case sensitive (preserved & matched exactly) |
| Mutability | Immutable after process instance creation |

Validation failures during process instance creation (too many tags, invalid pattern/length, duplicates) cause the create request to be rejected with a 4xx error; consequently no jobs will carry those tags.

### Use cases

- Routing and prioritization (e.g. `priority:high`).
- Business / domain identifiers from internal or third‑party systems (e.g. `businessKey:1234`, `customerId:7890`, `orderId:4567`).
- Cross-system correlation keys without exposing full variable payloads.
- Analytics segmentation (e.g. `region:emea`, `channel:web`).
- Feature rollout / experiment grouping (e.g. `experiment:checkout-v2`).
- Environment or tenant-like labeling where full multi-tenancy isn’t required (e.g. `env:staging`).

### Guidelines

- Do not store secrets or PII; tags propagate with jobs and exports.
- Prefer concise `key:value` or `key` patterns for consistency.

> - Use variables (not tags) for mutable or large data.

Reviewer 2: 👍 I like this!
